katka/views.py

```diff
@@ -230,26 +228,10 @@ def get_user_restricted_queryset(self, queryset):
         )
 
 
-def pre_validate_steprun_update(serializer):
-    """In case of an update to the status which sets the step to a final status, ensure the 'ended_at'
-       field is set.
-    """
-    if (
-        "status" in serializer.validated_data
-        and serializer.validated_data["status"] in STEP_FINAL_STATUSES
-        and "ended_at" not in serializer.validated_data
-    ):
-        serializer.validated_data["ended_at"] = datetime.now(tz=pytz.timezone(settings.TIME_ZONE))
-
-
```

Comment:
since this is now done in katka-pipeline-runner (and it makes more sense there), we can remove it here and make the code simpler